linode/manager
# upcoming: [M3-7618] - Delete Placement Group Modal

## Description 📝
This PR adds the delete placement group modal to the PG flow. It also touches a few components for some cleanup or enhancing (non breaking changes)

ℹ️ **CONTEXT**: The reason for adding the removable list in the delete modal is because (not unlike buckets) we can't delete a Placement Group that still has Linodes assigned to it

## Changes  🔄
- Add the PG delete modal and its test
- Make sure inputProps are passed in `TypeToConfirmDialog `
- Allow RemovableSelectionsList to take a readable remove label + test update + story

## Preview 📷
![Screenshot 2024-02-15 at 16 29 03](https://github.com/linode/manager/assets/130582365/1fbb8ef9-18e6-4b36-a097-04893090cacd)

## How to test 🧪

⚠️ NOTE: this is MSW mock data so we won't get the proper UI data changes 👎 

### Prerequisites
- Have the "Placement Group" flag and MSW turned on

### Verification steps 
- Navigate to `http://localhost:3000/placement-groups`
- Click "Delete" for any placement group
- Confirm UI and MSW requests
- change the [factory data](https://github.com/linode/manager/blob/6adb375576d89c9c630547349d1f0ce8092de295/packages/manager/src/factories/placementGroups.ts#L15) to an enabled delete modal

## As an Author I have considered 🤔

*Check all that apply*
